### PR TITLE
[DET-2974] perf: clean up master routes for webui

### PR DESCRIPTION
**Pull request by hamidzr**
## TODO
- [x] clean up the commits

### CODE REVIEW

**Comment by hamidzr** [comment]
we might wanna leave this for after https://github.com/determined-ai/determined/pull/300

**Comment by hamidzr** [comment]
once we have only one SPA and move serving UI from `/` we can do away with a lot of these one to one mappings for web assets.

File: master/internal/core.go [hunk 1]
```diff
@@ -484,23 +484,37 @@ func (m *Master) Run() error {
...
+		{"/favicon.ico", "favicon.ico"},
+		{"/favicon.ico", "favicon.ico"},
+		{"/det/*", "index.html"},
+	}
+
+	for _, fileRoute := range reactFiles {
+		m.echo.File(fileRoute.route, filepath.Join(reactRoot, fileRoute.path))
+	}
+
+	m.echo.Static("/favicons", filepath.Join(webuiRoot, "react/favicons"))
```
Comment (dzhu) [inline master/internal/core.go:515]: blocking: Use `reactRoot` for these too.

File: master/internal/core.go [hunk 2]
```diff
@@ -484,23 +484,37 @@ func (m *Master) Run() error {
 	})
 
 	// Elm WebUI.
-	m.echo.File("/", filepath.Join(webuiRoot, "elm/public/index.html"))
 	m.echo.File("/ui", filepath.Join(webuiRoot, "elm/public/index.html"))
```
Comment (dzhu) [inline master/internal/core.go:487]: blocking: Expand the same structure to the Elm routes and `Static` calls.

File: master/internal/core.go [hunk 3]
```diff
@@ -484,23 +484,37 @@ func (m *Master) Run() error {
 	})
```
Comment (dzhu) [inline master/internal/core.go:484]: non-blocking: Could you change the 301 in the line above this to `http.StatusMovedPermanently`? Slightly off-topic, perhaps, but I don't feel like making a PR just for that.

File: master/internal/core.go (hunk 2 above)
Comment (hamidzr) [reply master/internal/core.go]: was thinking maybe also mix some of the initialization but separate looks a bit nicer.

**Comment by ybt195** [comment]
@hamidzr Can you add some context as to how much of a performance improvement we get?

**Comment by hamidzr** [comment]
> @hamidzr Can you add some context as to how much of a performance improvement we get?

the performance part is a 2 line change the rest is just some cleanup. main change is about removing a redirect. currently, visitin `/` loads Elm first which then redirects to the react app. with this change we direct `/` to react.

**Comment by hamidzr** [comment]
about how much.. I was gonna try `time curl` but since the redirects are javascript based it wouldn't be as easy to time.

**Pull request merged**